### PR TITLE
seccomp: added Specification for seccomp backend

### PR DESCRIPTION
**Pull request by stolowski**
Added spec for seccomp - one of the many steps to migrate to new API. This just adds the spec, it's not used yet.

### CODE REVIEW

File: interfaces/seccomp/spec.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+// Implementation of methods required by interfaces.Specification
+
+// AddConnectedPlug records seccomp-specific side-effects of having a connected plug.
+func (spec *Specification) AddConnectedPlug(iface interfaces.Interface, plug *interfaces.Plug, slot *interfaces.Slot) error {
+	type definer interface {
+		SecCompConnectedPlug(spec *Specification, plug *interfaces.Plug, slot *interfaces.Slot) error
+	}
+	if iface, ok := iface.(definer); ok {
```
Comment (mvo5) [inline interfaces/seccomp/spec.go]: Maybe we could invert the logic here?
```
 +	if iface, ok := iface.(definer); !ok { return nil }
```
?

File: interfaces/seccomp/spec.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		spec.securityTags = plug.SecurityTags()
+		defer func() { spec.securityTags = nil }()
```
Comment (mvo5) [inline interfaces/seccomp/spec.go]: Why the defer here? Feels strange to me. Also looks dangerous for concurrent code. Maybe SecCompConnectedPlug() should have an additional securityTag paremeter or something?

Comment (stolowski) [inline interfaces/seccomp/spec.go]: This is to reset securityTags after the call to interface-supplied function (the "definer") finishes. There is no concurrency involved around this code.

File: interfaces/seccomp/spec.go (hunk 1 above)
Comment (stolowski) [inline interfaces/seccomp/spec.go]: Ok, I can do it if it makes it more readable.

**Comment by stolowski** [comment]
Closing in favor of #2924, will address one of @mvo5 comments there.

File: interfaces/seccomp/spec.go (hunk 1 above)
Comment (stolowski) [inline interfaces/seccomp/spec.go]: Ah, sorry, I think it needs to stay this way as I need the type of iface.(definer) to actually call the method of the "definer" interface.

**Pull request closed without merging**